src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperRunner.cs

```diff
@@ -46,12 +48,34 @@ public class TagHelperRunner
                 TagMode = executionContext.TagMode,
             };
 
-            foreach (var tagHelper in orderedTagHelpers)
+            foreach (var tagHelper in executionContext.TagHelpers)
             {
                 await tagHelper.ProcessAsync(tagHelperContext, tagHelperOutput);
             }
 
             return tagHelperOutput;
         }
+
+        private static void OrderTagHelpers(IList<ITagHelper> tagHelpers)
+        {
+            if (tagHelpers.Count == 1)
+            {
+                return;
+            }
+
+            ITagHelper temp = null;
+            for (var i = 0; i < tagHelpers.Count; i++)
+            {
+                for (var j = i + 1; j < tagHelpers.Count; j++)
+                {
+                    if (tagHelpers[j].Order < tagHelpers[i].Order)
+                    {
+                        temp = tagHelpers[i];
+                        tagHelpers[i] = tagHelpers[j];
+                        tagHelpers[j] = temp;
+                    }
+                }
+            }
```

Comment:
Doubt this is efficient for mid-sized collections but agree it's good enough for the small collections we expect. Add a comment about not spending a lot of time choosing the algorithm.